pandas/core/frame.py

```diff
@@ -9659,6 +9660,19 @@ def apply(
         args : tuple
             Positional arguments to pass to `func` in addition to the
             array/series.
+        by_row : False or "compat", default "compat"
+            If "compat", will if possible first translate the func into pandas
+            methods (e.g. ``Series().apply(np.sum)`` will be translated to
+            ``Series().sum()``). If that doesn't work, will try call to apply again with
+            ``by_row=True`` and if that fails, will call apply again with
+            ``by_row=False``
+            If False, the funcs will be passed the whole Series at once.
+            ``by_row`` only has effect when ``func`` is a listlike or dictlike of funcs
+            and the func isn't a string.
+            ``by_row=True`` has not been implemented, and will raise an
+            ``NotImplenentedError``.
```

Comment:
I think it'd be good to have the callout on this only applying to list/dict-likes at the beginning, and adding in that this is compatible with previous versions. What do you think about being more vague about the compat behavior instead of trying to detail it out? Something like

```
        by_row : False or "compat", default "compat"
            Only has effect an when ``func`` is a listlike or dictlike of funcs
            on the values that aren't NumPy functions (e.g. ``np.sum``) or 
            string-aliases for operations (e.g. ``"sum"``). 
            "compat" is backwards compatible with previous versions and will
            sometimes operate by row and sometimes operate on the whole Series at once.
            If False, the funcs will be passed the whole Series at once.
```

I'm also okay with keeping the more detailed description of compat if you prefer.

Reply:
My preference is for the other version, if that's ok. I changed it a bit though.